### ISSUE TITLE
IPV6 DNAT incorrectly quotes port number

### ISSUE THREAD

**Issue by torbennehmer** [fwbuilder/fwbuilder#66]
I am trying to DNAT a public IPv6 Address to an ULA address used internally by a docker container. During that process I also want to rewrite the destination port. If I do so, FWBuilder 5.3.7 forgets to quote the IPv6 address within brackets and thus ip6tables takes the :port suffix as part of the IP address:

Assume this public IP:
![image](https://user-images.githubusercontent.com/2812067/47044195-d3476480-d18f-11e8-8cef-13b6bf8e6dee.png)

... and this internal one:
![image](https://user-images.githubusercontent.com/2812067/47044222-e5290780-d18f-11e8-9d93-7c7e9c8f3d99.png)

Now I use a rule like this:
![image](https://user-images.githubusercontent.com/2812067/47044288-11448880-d190-11e8-88b0-aa869ae4b7f0.png)

This compiles into this rule:
`$IP6TABLES -t nat -A PREROUTING -i eth0  -p tcp -m tcp   -d 2001:...e::14:9  --dport 10101 -j DNAT --to-destination fde9:...46::2:22`

The last :22 is actually a port number, however, this is obviously ambigous in this context, ip6tables actually creates a rule without a port change but with an incorrect dest IP.

The correct rule would have to look like this (note the square brackets):
`$IP6TABLES -t nat -A PREROUTING -i eth0  -p tcp -m tcp   -d 2001:...e::14:9  --dport 10101 -j DNAT --to-destination [fde9:...46::2]:22`

**Comment by rephlex** [comment]
OT but can't resist: PLEASE don't NAT ipv6.

If you feel the need to do so, you are doing it wrong. It is discouraged by all current writings and the IETF.

**Comment by torbennehmer** [comment]
@rephlex I cannot agree more - the NAT in my case most definitly is not a false sense of security (by obscurity) or whatever. As a guy having studied computer networking I'd love to do it the one true way routing everything together. However, in this case I'm not at a leisure to do so. :-(

It solves the main problem, that I do not get any IPv6 address space routed to me, I don't even have a full /64 available (don't ask..., have do with what I've got). So I am using an ULA (actually two) internally to have a routed network, one for a few fully virtualized boxes, another one for an internal docker network spread out behind the fwbuilder Firewall. To make this work I do have to map everything to the outside.

As well, I have a second case where I need different port translation rules between an IPv4 forwarding and an IPv6 forwarding in a few special docker cases, which is where I stumbled on the mentioned bug. Even with a route net, I'd have to translate the port in that case - all other rules are 1:1 IP mappings which work like a breeze. But the Port translation done for other reasons fails.

PS: Schöne Grüße aus der Gegend von Ulm/Günzburg (Nähe Legoland)! :-)
